includes/class-wc-stripe-helper.php

```diff
@@ -181,8 +181,13 @@ public static function get_stripe_amount( $total, $currency = '' ) {
 			$currency = get_woocommerce_currency();
 		}
 
-		if ( in_array( strtolower( $currency ), self::no_decimal_currencies() ) ) {
+		$currency = strtolower( $currency );
+
+		if ( in_array( $currency, self::no_decimal_currencies(), true ) ) {
 			return absint( $total );
+		} elseif ( in_array( $currency, self::tree_decimal_currencies(), true ) ) {
+			$amount = absint( wc_format_decimal( ( (float) $total * 100 ), wc_get_price_decimals() ) ); // For tree decimal currencies.
+			return $amount - ( $amount % 10 ); // Round the last digit down. See https://docs.stripe.com/currencies?presentment-currency=AE#three-decimal
```

Comment:
Is this calculation correct?

Using the example from the stripe docs:

> For example, 5.124 KWD must be rounded to 5120 or 5130.

When I test this amount using the following code: `WC_Stripe_Helper::get_stripe_amount( 5.124, 'KWD' );`, it's returning 510. Not 5120 according to Stripe.

Is this right?

Reply:
I figured out the problem while writing my review 😄 

The issue is the `* 100` when you have the **Number of decimals** setting found under **WooCommerce > Settings > General** set to 3 (instead of 2).

To fix this I think we need to do something like:

```
$cent_precision = pow( 10, wc_get_price_decimals() );
$amount         = absint( wc_format_decimal( ( (float) $total * $cent_precision ), wc_get_price_decimals() ) );
```

Reply:
Great catch! I forgot about this setting, sorry! I have included your suggestion and also updated the unit tests 👍

Reply:
It stopped working when the settings were configured for two decimal places. I am looking 👀

Reply:
Oh OK, hardcoding the decimal multiplier to `1000` did the trick. Working for me with both decimal places settings. Can you repeat your tests as well please, @mattallan?

Reply:
Nice find and thanks for adding a unit test to this one!

> hardcoding the decimal multiplier to 1000 did the trick. Working for me with both decimal places settings.

Interesting! I didn't think that would be the solution so I'm sorry for the bum steer with using the `pow()` calculation.

I've tested the latest and confirmed it's working as expected with your changes 💃

Reply:
No worries! I couldn't find the bug with the three decimal settings if it weren't for you 🙌